### PR TITLE
feat: Add Traefik gateway container CPU & RAM request/limit resources

### PR DESCRIPTION
**Pull request by mshaposhnik**
Signed-off-by: Max Shaposhnik <mshaposh@redhat.com>

### What does this PR do?
Adds Traefic gateway container CPU & RAM request/limit resources

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20520

### How to test this PR?
Run devworkspace mode, check Traefic container limits.

image

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by sparkoo — APPROVED** [review]

File: controllers/devworkspace/solver/che_routing.go [hunk 1]
```diff
@@ -162,13 +163,23 @@ func (c *CheRoutingSolver) provisionPodAdditions(objs *solvers.RoutingObjects, c
 	objs.PodAdditions.Containers = append(objs.PodAdditions.Containers, corev1.Container{
 		Name:            wsGatewayName,
 		Image:           cheCluster.Spec.Gateway.Image,
-		ImagePullPolicy: corev1.PullAlways,
+		ImagePullPolicy: corev1.PullIfNotPresent,
```
Comment (skabashnyuk) [inline controllers/devworkspace/solver/che_routing.go:166]: why do we need this change?

**Review by skabashnyuk — APPROVED** [review]
lgtm in general

File: controllers/devworkspace/solver/che_routing.go (hunk 1 above)
Comment (mshaposhnik) [reply controllers/devworkspace/solver/che_routing.go:166]: to avoid image pull on each and every WS startup

Comment (tolusha) [reply controllers/devworkspace/solver/che_routing.go:166]: We can use `deploy.DefaultPullPolicyFromDockerImage()` to detect Image pull policy.

**Review by tolusha — APPROVED** [review]

**Pull request merged**